### PR TITLE
Scrolling nested containers

### PR DESCRIPTION
**Pull request by darrenburns**
In the video below you can see me pressing a keybind to scroll to a nested (grandchild) widget.

https://user-images.githubusercontent.com/5740731/170022543-33d3cabb-daf4-4d3f-bc60-950848353d47.mov

### CODE REVIEW

File: sandbox/basic.css [hunk 1]
```diff
@@ -69,14 +69,14 @@ App > Screen {
...
   background: $background;
   layout: vertical;
-  overflow-y: scroll; 
+  overflow-y: scroll;
 }
 
 
 Tweet {
-  height: auto;
+  height: 10;
```
Comment (DrBenton) [inline sandbox/basic.css:77]: this poor CSS is constantly updated by us in every direction :sweat_smile:

File: src/textual/widget.py [hunk 2]
```diff
@@ -514,38 +514,50 @@ def scroll_to_widget(self, widget: Widget, *, animate: bool = True) -> bool:
...
-            delta_y = min(top_delta.y, bottom_delta.y, key=abs)
+            if node == self:
+                break
+            child = node
+            node = node.parent
 
-        return self.scroll_relative(
-            delta_x or None, delta_y or None, animate=animate, duration=0.2
-        )
+        # TODO: Return boolean value
```
Comment (DrBenton) [inline src/textual/widget.py:560]: shouldn't this TODO be addressed before we merge the PR?
(unless it's a complicated value to compute with this new version of the method, of course :-)

**Review by DrBenton — COMMENTED** [review]
Wow, that was quick! Nice one :blush:

**Comment by darrenburns** [comment]
@willmcgugan Could you have a look at this when you get a chance? It's essentially just the same code as before but inside a loop, applying going up the DOM.

It also updates it such that it's the whole region of the widget that scrolls into view, rather than just the content region.

**Review by willmcgugan — APPROVED** [review]
LGTM. I think you said there were some awkward edge cases not yet handled?

**Pull request merged**